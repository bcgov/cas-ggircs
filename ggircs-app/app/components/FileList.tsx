import React, { useState } from "react";
import { Row, Col, Table } from "react-bootstrap";

const data = [
  {
    created_at: "Thu, 14 May 2020 05:25:43 GMT",
    name: "GHGBC_PROD_20120629.zip",
    size: 69.45426750183105,
  },
  {
    created_at: "Thu, 14 May 2020 05:31:32 GMT",
    name: "GHGBC_PROD_20121130.zip",
    size: 290.5470142364502,
  },
  {
    created_at: "Thu, 14 May 2020 05:31:30 GMT",
    name: "GHGBC_PROD_20121214.zip",
    size: 290.45326709747314,
  },
  {
    created_at: "Thu, 14 May 2020 05:30:22 GMT",
    name: "GHGBC_PROD_20130510.zip",
    size: 234.48462772369385,
  },
];

const FileList: React.FunctionComponent = () => {
  const [selectedFile, setSelectedFile] = useState("");

  return (
    <div>
      <Row>
        <Col md="4">
          <Table size="sm" borderless striped hover>
            <thead>
              <tr>
                <th>File</th>
                <th>Date</th>
                <th>Size</th>
              </tr>
            </thead>
            <tbody>
              {data.map((element) => (
                <tr
                  key={element.name}
                  onClick={() => setSelectedFile(element.name)}
                >
                  <td>{element.name}</td>
                  <td>{new Date(element.created_at).toDateString()}</td>
                  <td>{Math.round(element.size)}M</td>
                </tr>
              ))}
            </tbody>
          </Table>
        </Col>
        <Col md="8">
          <h4>{selectedFile}</h4>
          <p>details</p>
        </Col>
      </Row>
      <style jsx>
        {`
          td,
          th {
            font-size: 0.8em;
          }
        `}
      </style>
    </div>
  );
};

export default FileList;
