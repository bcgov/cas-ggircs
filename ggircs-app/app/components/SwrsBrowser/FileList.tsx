import React, { useEffect, useState } from "react";
import { EcccFile } from "types";
import { Row, Col, ListGroup } from "react-bootstrap";
import FileDetails from "./FileDetails";

const FileList: React.FunctionComponent = () => {
  const [selectedFile, setSelectedFile] = useState<EcccFile>(null);
  const [files, setFiles] = useState<EcccFile[]>([]);

  useEffect(() => {
    (async () => {
      const filesResponse = await fetch("/api/eccc/files");
      setFiles(await filesResponse.json());
    })();
  }, []);

  return (
    <Row>
      <Col md={4}>
        <h2>Uploaded Files</h2>
        <div className="scrollable">
          <ListGroup>
            {files.map((file) => (
              <ListGroup.Item
                action
                onClick={() => setSelectedFile(file)}
                active={selectedFile && selectedFile.name === file.name}
                key={file.name}
              >
                {file.name}
              </ListGroup.Item>
            ))}
          </ListGroup>
        </div>
      </Col>
      <Col md={8}>
        {selectedFile && <FileDetails ecccFile={selectedFile} />}
      </Col>
      <style jsx>
        {`
          h2 {
            font-size: 20px;
          }
          .scrollable {
            max-height: calc(100vh - 400px);
            overflow-y: auto;
          }
        `}
      </style>
    </Row>
  );
};

export default FileList;
