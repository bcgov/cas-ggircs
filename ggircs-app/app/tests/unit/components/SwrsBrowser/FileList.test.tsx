import React from "react";
import { EcccFile } from "next-env";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import FileList from "components/SwrsBrowser/FileList";

const ecccFiles: EcccFile[] = [
  {
    created_at: "Thu, 14 May 2020 05:25:43 GMT",
    name: "a.zip",
    size: 69.45,
  },
  {
    created_at: "Thu, 14 May 2020 05:31:32 GMT",
    name: "b.zip",
    size: 290.54,
  },
  {
    created_at: "Thu, 14 May 2020 05:31:30 GMT",
    name: "c.zip",
    size: 290.45,
  },
];

const mockFetch = async (url: string) => {
  if (url === "/api/eccc/files") {
    return {
      json: async () => ecccFiles,
    };
  }
};

describe("The FileList component", () => {
  beforeAll(() => {
    global.fetch = jest.fn().mockImplementation(mockFetch);
  });

  afterAll(() => {
    global.fetch.mockClear();
    delete global.fetch;
  });

  it("should load the list of files", async () => {
    const component = await mount(<FileList />);
    await act(async () => {
      //
      await new Promise((resolve) => setImmediate(resolve));
      component.update();
    });
    expect(component).toMatchSnapshot();
    expect(component.find("button.list-group-item")).toHaveLength(3);
  });
});
