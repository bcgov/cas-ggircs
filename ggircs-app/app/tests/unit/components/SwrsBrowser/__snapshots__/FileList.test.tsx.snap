// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FileList component should load the list of files 1`] = `
<div
  className="row"
>
  <div
    className="col-md-4"
  >
    <h2
      className="jsx-3437169214"
    >
      Uploaded Files
    </h2>
    <div
      className="jsx-3437169214 scrollable"
    >
      <div
        className="list-group"
        onKeyDown={[Function]}
      >
        <button
          className="list-group-item list-group-item-action"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          a.zip
        </button>
        <button
          className="list-group-item list-group-item-action"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          b.zip
        </button>
        <button
          className="list-group-item list-group-item-action"
          data-rb-event-key={null}
          disabled={false}
          onClick={[Function]}
        >
          c.zip
        </button>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
  />
</div>
`;
